From: jbeulich@suse.com
Subject: netback: correct array index
References: bsc#983348
Patch-mainline: Never, SUSE-Xen specific

Since the array is being populated backwards, accessing the previously
filled slot requires a positive index, not a negative one.

--- a/drivers/xen/netback/netback.c
+++ b/drivers/xen/netback/netback.c
@@ -1170,7 +1170,7 @@ static void netbk_fill_tx_copy(const net
 		gop->ptr += first;
 
 		gop->copy--;
-		gop->copy->source = gop->copy[-1].source;
+		gop->copy->source = gop->copy[1].source;
 		gop->copy->source.offset += first;
 		gop->copy->dest.u.gmfn = virt_to_mfn(gop->ptr);
 		gop->copy->dest.domid = DOMID_SELF;
